viewing_party/party.py

```diff
@@ -0,0 +1,204 @@
+
+def get_friends_unique_watched(user_data):
+    watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+
+    user_data_list = []
+    friends_watched_list_pre = []
+    friends_watched_list_final = []
+   
+    unique_movie_list_friends = []
+   
+    for i in range(len(watched)):
+        user_data_list.append(watched[i])
+   
+    for j in range(len(friends_watched)):
+        friends_watched_list_pre.append((friends_watched[j]["watched"]))
+   
+    for k in range(len(friends_watched_list_pre)):
+        friends_watched_list_final += friends_watched_list_pre[k]
+
+    friends_list_no_duplicates = []
+    for i in friends_watched_list_final:
+        if i not in friends_list_no_duplicates:
+            friends_list_no_duplicates.append(i)
+           
+    for movie in friends_list_no_duplicates:
+        if not movie in user_data_list:
+            unique_movie_list_friends.append(movie)
+
+    return unique_movie_list_friends
+
+
+
+# WAVE 4
+def get_available_recs(user_data):
+    pre_recommendations = []
+    
+    final_recommendations = []
+    friends_movies = user_data["friends"]
+    friends_watched_list = []
+    user_watched_movies = user_data["watched"]
+
+    for friends_watched_dict in friends_movies:
+        for key, value in friends_watched_dict.items():
+            for movie in value:
+                friends_watched_list.append(movie)
+
+   
+    for recommendation in friends_watched_list:
+      for subscription in user_data["subscriptions"]:
+        if recommendation["host"] == subscription:
+          pre_recommendations.append(recommendation)
+
+    for unique_movie in pre_recommendations:
+      if unique_movie not in final_recommendations:
+        final_recommendations.append(unique_movie)
+    
+    if user_watched_movies: 
+      for movie in user_watched_movies:
+        for recom in final_recommendations: 
+          if movie["title"] in recom["title"]:
+            final_recommendations = []       
+          else:
+            final_recommendations = []
+            final_recommendations.append(recom)
+   
+    return final_recommendations
+
+
+
+# WAVE 5
+def get_new_rec_by_genre(user_data):
+    most_frequent = get_most_watched_genre(user_data)
+    #print(type(most_frequent))
+    
+    recommendations = []
+    recommendations_final = []
+    friends_movies = user_data["friends"]
+    friends_watched_list = []
+    user_watched_movies = user_data["watched"]
+
+    for friends_watched_dict in friends_movies:
+        for key, value in friends_watched_dict.items():
+            for movie in value:
+                friends_watched_list.append(movie)
+    # print(friends_watched_list)
+    if user_watched_movies:
+      for friends_movie in friends_watched_list:
+        for movie in user_watched_movies:
+          if not friends_movie["title"] == movie["title"]:
+            if friends_movie["genre"] == most_frequent:
+              recommendations.append(friends_movie)
```

Comment:
There is a logical bug here.  A `friends_movie` will be compared to every movie in the user_watched_movies, and if it's not equal to _any_ of them, and the genre is the most_frequent, it'll be added to recommendations.  This can result in adding a movie that is in the user_data["watched"] list to the recommendation.  A movie should be added to the recommendations only when it's not equal to _all_ of the movies in the user_watched_movies.